android/src/main/java/com/swmansion/rnscreens/bottomsheet/SheetDelegate.kt

```diff
@@ -237,6 +250,30 @@ class SheetDelegate(
         }
     }
 
+    internal fun calculateSheetOffsetY(keyboardHeight: Int): Int {
+        val containerHeight = tryResolveContainerHeight()
+        check(containerHeight != null) {
+            "[RNScreens] Failed to find window height during bottom sheet behaviour configuration"
+        }
+
+        if (screen.isSheetFitToContents()) {
+            val contentHeight = screen.contentWrapper?.height ?: 0
+            val offsetFromTop = containerHeight - contentHeight
+            return minOf(offsetFromTop, keyboardHeight)
+        }
+
+        val detents = screen.sheetDetents
+        if (detents.isEmpty()) {
+            throw IllegalStateException("[RNScreens] Cannot determine sheet detent - detents list is empty")
+        }
```

Comment:
We should have separate data structure for detents, making sure of this invariant. Let's create ticket for this.

Reply:
https://github.com/software-mansion/react-native-screens-labs/issues/536